compiler+runtime/src/cpp/jank/runtime/obj/lazy_sequence.cpp

```diff
@@ -155,8 +161,10 @@ namespace jank::runtime::obj
 
   lazy_sequence_ptr lazy_sequence::with_meta(object_ptr const m) const
   {
+    resolve_seq();
+    auto const s(sequence ? runtime::fresh_seq(sequence) : nil::nil_const());
+    auto const ret(make_box<lazy_sequence>(nullptr, s == nil::nil_const() ? nullptr : s));
```

Comment:
There is no need for freshness here. Also, we're doing an odd dance where we turn `s` into either a fresh seq or `nil`, then we do another ternary to remove the `nil`. Why not just remove the `nil` part altogether?

```cpp
    auto const s(sequence ? runtime::fresh_seq(sequence) : nullptr);
    auto const ret(make_box<lazy_sequence>(nullptr, s));
```

Then remove the freshness altogether.

```cpp
    auto const ret(make_box<lazy_sequence>(nullptr, sequence));
```

Reply:
Much nicer. The odd dance was because `runtime::fresh_seq(sequence)` returns `nil`.

Reply:
Ah, that makes sense now.